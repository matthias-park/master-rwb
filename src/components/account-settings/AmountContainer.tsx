import React from 'react';

const AmountContainer = ({ title, amount }) => {
  return (
    <div className="amount-container mb-4">
      <small className="amount-container__text">{title}</small>
      <h2 className="amount-container__amount">â‚¬ {amount}</h2>
      <i className="amount-container__icon icon-tooltip"></i>
    </div>
  )
}

export default AmountContainer;