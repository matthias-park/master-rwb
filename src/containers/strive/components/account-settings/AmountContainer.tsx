import React from 'react';
import OverlayTrigger from 'react-bootstrap/OverlayTrigger';
import Tooltip from 'react-bootstrap/Tooltip';
import clsx from 'clsx';
interface Props {
  title: string;
  amount: string | number;
  tooltip?: string;
}

const AmountContainer = ({ title, amount, tooltip }: Props) => {
  return (
    <div className="amount-container mb-4">
      <small data-testid="amount-title" className="amount-container__text">
        {title}
      </small>
      <h2 data-testid="amount-amount" className="amount-container__amount">
        {amount}
      </h2>
      {!!tooltip && (
        <OverlayTrigger
          placement={'bottom'}
          overlay={<Tooltip id="tooltip_amount">{tooltip}</Tooltip>}
        >
          <i
            data-testid="amount-tooltip"
            className={clsx(
              `icon-${window.__config__.name}-tooltip`,
              'amount-container__icon',
            )}
          />
        </OverlayTrigger>
      )}
    </div>
  );
};

export default AmountContainer;
