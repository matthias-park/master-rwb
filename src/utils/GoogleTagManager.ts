import {
  IDataLayer,
  ISnippets,
  ISendToGTM,
  ISetupGTM,
  ISnippetsParams,
} from '../types/GoogleTagManager';

const getDataLayerSnippet = (
  dataLayer: Pick<IDataLayer, 'dataLayer'>['dataLayer'],
  dataLayerName: Pick<
    IDataLayer,
    'dataLayerName'
  >['dataLayerName'] = 'dataLayer',
): Pick<ISnippets, 'gtmDataLayer'>['gtmDataLayer'] =>
  `window.${dataLayerName} = window.${dataLayerName} || []; window.${dataLayerName}.push(${JSON.stringify(
    dataLayer,
  )})`;

const getIframeSnippet = (
  id: Pick<ISnippetsParams, 'id'>['id'],
  environment?: Pick<ISnippetsParams, 'environment'>['environment'],
) => {
  let params = ``;
  if (environment) {
    const { gtm_auth, gtm_preview } = environment;
    params = `&gtm_auth=${gtm_auth}&gtm_preview=${gtm_preview}&gtm_cookies_win=x`;
  }
  return `<iframe src="https://www.googletagmanager.com/ns.html?id=${id}${params}" height="0" width="0" style="display:none;visibility:hidden" id="tag-manager"></iframe>`;
};

const getGTMScript = (
  dataLayerName: Pick<ISnippetsParams, 'dataLayerName'>['dataLayerName'],
  id: Pick<ISnippetsParams, 'id'>['id'],
  environment?: Pick<ISnippetsParams, 'environment'>['environment'],
) => {
  let params = ``;
  if (environment) {
    const { gtm_auth, gtm_preview } = environment;
    params = `+"&gtm_auth=${gtm_auth}&gtm_preview=${gtm_preview}&gtm_cookies_win=x"`;
  }
  return `
      (function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start':
        new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0],
        j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src=
        'https://www.googletagmanager.com/gtm.js?id='+i+dl${params};f.parentNode.insertBefore(j,f);
      })(window,document,'script','${dataLayerName}','${id}');
    `;
};
/**
 * Function to setup the Google Tag Manager
 * @param params - The snippets params
 */
const setupGTM = (params: ISnippetsParams): ISetupGTM => {
  const getDataLayerScript = (): HTMLElement => {
    const dataLayerScript = document.createElement('script');
    dataLayerScript.innerHTML = getDataLayerSnippet(
      params.dataLayer,
      params.dataLayerName,
    );
    return dataLayerScript;
  };

  const getNoScript = (): HTMLElement => {
    const noScript = document.createElement('noscript');
    noScript.innerHTML = getIframeSnippet(params.id, params.environment);
    return noScript;
  };

  const getScript = (): HTMLElement => {
    const script = document.createElement('script');
    script.innerHTML = getGTMScript(
      params.dataLayerName,
      params.id,
      params.environment,
    );
    return script;
  };

  return {
    getDataLayerScript,
    getNoScript,
    getScript,
  };
};

/**
 * Function to init the GTM
 * @param dataLayer - The dataLayer
 * @param dataLayerName - The dataLayer name
 * @param environment - Specify the custom environment to use
 * @param id - The ID of the GTM
 */
export const initGTM = ({
  dataLayer,
  dataLayerName,
  environment,
  id,
}: ISnippetsParams): void => {
  const gtm = setupGTM({
    dataLayer,
    dataLayerName,
    environment,
    id,
  });

  const dataLayerScript = gtm.getDataLayerScript();
  const script = gtm.getScript();
  const noScript = gtm.getNoScript();

  document.head.insertBefore(script, document.head.childNodes[0]);
  document.head.insertBefore(dataLayerScript, document.head.childNodes[0]);
  document.body.insertBefore(noScript, document.body.childNodes[0]);
};

/**
 * Function to send the events to the GTM
 * @param dataLayerName - The dataLayer name
 * @param data - The data to push
 */
export const sendToGTM = ({ dataLayerName, data }: ISendToGTM): void =>
  window[dataLayerName].push(data);
