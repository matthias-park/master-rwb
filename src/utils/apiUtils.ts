import fetch from 'isomorphic-unfetch';
import { ConfigInterface, mutate } from 'swr';
import { RailsApiResponseFallback } from '../constants';
import RailsApiResponse from '../types/api/RailsApiResponse';
import { RegistrationPostalCodeAutofill } from '../types/api/user/Registration';
import { cleanPostBody } from '.';
import * as Sentry from '@sentry/react';

export const formatUrl = (
  url: string,
  params: { [key: string]: string | number | undefined } = {},
): string => {
  const paramsArr = Object.keys(params)
    .filter(key => params[key])
    .map(key => `${key}=${params[key]}`);
  return `${url}${paramsArr.length ? '?' : ''}${paramsArr.join('&')}`;
};

interface GetApiOptions {
  responseText?: boolean;
  cache?: RequestCache;
}

export const getApi = <T>(url: string, options?: GetApiOptions): Promise<T> => {
  const config: RequestInit = {
    mode: 'cors',
    credentials: 'include',
    cache: options?.cache,
    headers: new Headers(),
  };
  const getUrl = url.startsWith('http')
    ? url
    : `${window.__config__.apiUrl}${url}`;
  return fetch(getUrl, config).then(res => {
    if (!res.ok && res.status !== 400) {
      if (![401, 403].includes(res.status)) {
        Sentry.captureMessage(
          `Request failed ${url} with status ${res.status}`,
          Sentry.Severity.Fatal,
        );
      } else {
        mutate('/restapi/v1/user/status');
      }
      return Promise.reject<RailsApiResponse<null>>({
        ...RailsApiResponseFallback,
        Code: res.status,
      });
    }
    if (options?.responseText) {
      return res.text();
    }
    return res.json();
  });
};

export interface PostOptions {
  formData?: boolean;
}

export const postApi = <T>(
  url: string,
  body?: {
    [key: string]: number | string | Blob | boolean | undefined | null | object;
  },
  options: PostOptions = {},
): Promise<T> => {
  const config: RequestInit = {
    method: 'post',
    mode: 'cors',
    credentials: 'include',
    headers: new Headers(),
  };
  if (body) {
    if (options.formData) {
      const formData = new FormData();
      const formBody = cleanPostBody(body);
      for (const key in formBody) {
        if (Array.isArray(formBody[key])) {
          for (const item of formBody[key]) {
            formData.append(key, item as string | Blob);
          }
        } else {
          formData.append(key, formBody[key] as string | Blob);
        }
      }
      config.body = formData;
    } else {
      config.body = JSON.stringify(cleanPostBody(body));
      (config.headers as Headers).append('Content-Type', 'application/json');
    }
  }
  const postUrl = url.startsWith('http')
    ? url
    : `${window.__config__.apiUrl}${url}`;
  return fetch(postUrl, config).then(res => {
    if (!res.ok && res.status !== 400) {
      if (![401, 403].includes(res.status)) {
        Sentry.captureMessage(
          `Request failed ${url} with status ${res.status}`,
          Sentry.Severity.Fatal,
        );
      } else {
        mutate('/restapi/v1/user/status');
      }
      return Promise.reject<RailsApiResponse<null>>({
        ...RailsApiResponseFallback,
        Code: res.status,
      });
    }
    return res.json();
  });
};

export const SwrFetcherConfig: ConfigInterface<
  any,
  any,
  (...args: any) => any
> = {
  fetcher: getApi,
  revalidateOnFocus: false,
  revalidateOnReconnect: false,
  errorRetryCount: 2,
  dedupingInterval: 30000,
  focusThrottleInterval: 10000,
  errorRetryInterval: 10000,
  onError: (err, key) => {
    if (![401, 403].includes(err.status)) {
      Sentry.captureMessage(
        `Request failed ${key} with status ${err.status}`,
        Sentry.Severity.Fatal,
      );
    }
  },
};

export const formatSuccesfullRailsApiResponse = <T>(
  data: T,
): RailsApiResponse<T> => ({
  Code: 0,
  Data: data,
  Message: '',
  Success: true,
});

export const API_VALIDATIONS = {
  email: async (email: string): Promise<RailsApiResponse<{}>> => {
    const res = await postApi<RailsApiResponse<{}>>(
      '/restapi/v1/registration/check/email',
      {
        email,
      },
    ).catch(err => err);
    return res;
  },
  personalCode: async (
    personal_code: string,
  ): Promise<RailsApiResponse<{}>> => {
    const res = await postApi<RailsApiResponse<{}>>(
      '/restapi/v1/registration/check/personal_code',
      {
        personal_code,
      },
    ).catch((err: RailsApiResponse<{}>) => err);
    return res;
  },
  postalCode: async (
    post_code: string,
  ): Promise<RailsApiResponse<RegistrationPostalCodeAutofill | null>> => {
    const res = await postApi<
      RailsApiResponse<RegistrationPostalCodeAutofill | null>
    >('/restapi/v1/registration/check/post_code', {
      post_code,
    }).catch((err: RailsApiResponse<null>) => err);
    return res;
  },
};
