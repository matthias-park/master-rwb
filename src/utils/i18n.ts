import { ConfigRoute } from '../types/Config';
type Symbols = { [key: string]: { [key: string]: string } };

const i18n = () => {
  let locale = '';
  const symbols: Symbols = {};

  return {
    set(lang: string, data: unknown = {}) {
      symbols[lang] = Object.assign(symbols[lang] || {}, data);
    },

    locale(lang?: string) {
      return (locale = lang || locale);
    },

    table(lang: string) {
      return symbols[lang];
    },

    t(key: string, lang?: string) {
      const val = symbols?.[lang || locale]?.[key] || '';
      return val;
    },
  };
};

export const getRedirectLocalePathname = (
  availableLocales: string[],
  defaultLocale: string,
  availableRoutes: ConfigRoute[],
) => {
  const windowPaths = window.location.pathname.split('/');
  let urlLocale = defaultLocale;
  let urlPaths = window.location.pathname;

  if (availableLocales.includes(windowPaths[1])) {
    urlLocale = windowPaths[1];
  } else if (availableRoutes.some(route => route.path === urlPaths)) {
    urlPaths = `/${urlLocale}${urlPaths}`;
  } else {
    const urlPathWithoutFirstPath = urlPaths.replace(`/${windowPaths[1]}`, '');
    if (availableRoutes.some(route => route.path === urlPathWithoutFirstPath)) {
      urlPaths = `/${urlLocale}${urlPathWithoutFirstPath}`;
    } else {
      urlPaths = `/${urlLocale}`;
    }
  }

  if (window.location.pathname !== urlPaths) {
    window.history.replaceState({}, '', urlPaths);
  }
  return urlLocale;
};

export const setLocalePathname = (newlocale: string) => {
  const paths = window.location.pathname.split('/');
  paths[1] = newlocale;
  const newPath = paths.join('/');
  window.history.pushState({}, '', newPath);
};

export type I18n = ReturnType<typeof i18n>;
export default i18n;
