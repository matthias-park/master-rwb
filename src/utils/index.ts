import { PagesName } from '../constants';
import { PageConfig } from '../types/api/PageConfig';
import Lockr from 'lockr';
import { PostItem } from '../types/api/Posts';
import dayjs from 'dayjs';

export const sortAscending = (a: number, b: number) => a - b;
export const sortDescending = (a: number, b: number) => b - a;

export function throttle(func: Function, limit: number): Function {
  let inThrottle: boolean;

  return function (this: any): any {
    const args = arguments;
    const context = this;
    if (!inThrottle) {
      inThrottle = true;
      func.apply(context, args);
      setTimeout(() => (inThrottle = false), limit);
    }
  };
}

export const stringToMiliseconds = (time: string): number | null => {
  const parsed = time.split(':');
  if (parsed.length === 3) {
    return miliseconds(Number(parsed[0]), Number(parsed[1]), Number(parsed[2]));
  }
  return null;
};
export const miliseconds = (
  hrs: number = 0,
  min: number = 0,
  sec: number = 0,
): number => {
  return (hrs * 60 * 60 + min * 60 + sec) * 1000;
};
export const removeFalsyFromObject = (obj: any) =>
  Object.keys(obj).forEach(
    k => !obj[k] && obj[k] !== undefined && delete obj[k],
  );

export const structuredBankCommunications = (barcode): string => {
  if (!barcode) return '';
  const rExp = /\d+/;
  const moduloOf97 = barcode.match(rExp)[0] % 97;
  return '00' + barcode + moduloOf97.toString().padStart(2, '0');
};

export const formatNavigationRoutes = (constants: PageConfig) => {
  const navigationRoutes = constants.navigation_routes
    ? [...constants.navigation_routes]
    : [];
  if (constants.content_pages) {
    for (const [key, value] of Object.entries(constants.content_pages)) {
      const name = Array.isArray(value) ? value[0] : value;
      const slug = Array.isArray(value) ? value[1] : key;
      const contentPagePath = `${slug.startsWith('/') ? '' : '/'}${slug}`;
      if (!navigationRoutes.some(route => route.path === contentPagePath)) {
        navigationRoutes.push({
          id:
            Array.isArray(value) && PagesName[key]
              ? Number(key)
              : PagesName.TemplatePage,
          path: contentPagePath,
          name,
        });
      }
    }
  }
  return navigationRoutes.sort((a, b) =>
    sortDescending(a.path.length, b.path.length),
  );
};

export const clearUserLocalStorage = () => {
  const keys = ['transactions-date-to', 'transactions-date-from'];
  for (const key of keys) {
    localStorage.removeItem(key);
  }
};

export const cleanPostBody = (obj: { [key: string]: unknown }) =>
  Object.entries(obj).reduce((obj, [key, value]) => {
    if (!key.includes('temp_field_') && value != null) {
      obj[key] = typeof value === 'string' ? value.trim() : value;
    }
    return obj;
  }, {});

export const errorHandler = (event: ErrorEvent) => {
  if (event.error?.stack?.includes('kambi-chunk')) {
    const kambiErrorId = 'kambi-error-reload';
    const kambiErrorRetryCount = Lockr.get(kambiErrorId, 0) + 1;
    if (kambiErrorRetryCount < 3) {
      Lockr.set(kambiErrorId, kambiErrorRetryCount);
      window.location.reload();
    }
  }
};

export const filterPromotionsList = (promotions: PostItem[]): PostItem[] =>
  promotions.filter(promotion => {
    const isNotExpired =
      !promotion.expiration_date ||
      dayjs(promotion.expiration_date).isAfter(dayjs());
    const isPublished =
      promotion.publish_date && dayjs(promotion.publish_date).isBefore(dayjs());
    const isVisible = promotion.visible;
    return isNotExpired && isPublished && isVisible;
  });
