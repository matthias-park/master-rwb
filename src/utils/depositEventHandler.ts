import { CustomWindowEvents, ComponentName } from '../constants';
import RequestReturn from '../types/api/deposits/RequestReturn';
import RailsApiResponse from '../types/api/RailsApiResponse';
import depositEventHandlerProps from '../types/DepositEventHandler';
import { postApi } from './apiUtils';
import * as Sentry from '@sentry/react';

const depositEventHandler = ({
  eventType,
  eventData,
  setCustomHtml,
  depositStatus,
  setApiError,
  setDepositLoading,
  t,
  depositRequestId,
  bankId,
  locale,
  setCurrentStep,
  disableModal,
  depositForm,
}: depositEventHandlerProps) => {
  switch (eventType) {
    case CustomWindowEvents.DepositPaymentSuccess: {
      setApiError(null);
      setCustomHtml(null);
      depositStatus.startCheckingStatus();
      break;
    }
    case CustomWindowEvents.DepositPaymentError: {
      if (typeof eventData === 'string') {
        Sentry.captureMessage(`deposit error: ${eventData}`);
        setApiError({ message: eventData, variant: 'danger' });
      } else {
        setApiError({ message: t('api_response_failed'), variant: 'danger' });
      }
      setCustomHtml(null);
      setDepositLoading(false);
      break;
    }
    case CustomWindowEvents.DepositPaymentWarning: {
      if (typeof eventData === 'string') {
        setApiError({ message: eventData, variant: 'warning' });
      }
      break;
    }
    case CustomWindowEvents.DepositRequestReturn: {
      if (eventData && typeof eventData === 'object') {
        const data: any = {
          ...eventData,
          depositRequestId,
        };
        if (bankId === 189) {
          data.browserInfo = {
            acceptHeader: '*',
            colorDepth: window.screen.colorDepth,
            javaEnabled: false,
            language: locale,
            screenHeight: window.innerHeight,
            screenWidth: window.innerWidth,
            timeZoneOffset: new Date().getTimezoneOffset(),
            userAgent: window.navigator.userAgent,
          };
        }
        postApi<RailsApiResponse<RequestReturn>>(
          '/restapi/v1/deposits/request_return',
          {
            bank_id: bankId.toString(),
            data: JSON.stringify(data),
          },
        )
          .then(res => {
            if (res.Data.OK) {
              if (res.Data.AdditionalData?.redirectUrl) {
                if (depositForm) {
                  window.open(res.Data.AdditionalData.redirectUrl, '_blank');
                  disableModal?.(ComponentName.QuickDepositModal);
                } else {
                  window.location.href = res.Data.AdditionalData.redirectUrl;
                }
                return;
              }
              if (res.Data.AdditionalData?.verifyPaymentData) {
                return window.postMessage(
                  JSON.stringify({
                    eventType: CustomWindowEvents.DepositVerifyPayment,
                    eventData: res.Data.AdditionalData.verifyPaymentData,
                  }),
                  '*',
                );
              }
              setCustomHtml(null);
              depositStatus.startCheckingStatus();
            } else if (res.Data.Error) {
              setCurrentStep?.(1);
              setApiError({
                message: t('api_response_failed'),
                variant: 'danger',
              });
              Sentry.captureMessage(
                `Got DepositRequestReturn error: ${res.Data.Error}`,
                Sentry.Severity.Info,
              );
            }
          })
          .catch(() => null);
      } else {
        Sentry.captureMessage(
          'Got DepositRequestReturn without data',
          Sentry.Severity.Fatal,
        );
      }
    }
  }
};
export default depositEventHandler;
