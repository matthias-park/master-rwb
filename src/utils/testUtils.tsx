import React from 'react';
import { render } from '@testing-library/react';
import { SWRConfig } from 'swr';
import { SwrFetcherConfig } from './apiUtils';
import { ToastProvider } from 'react-toast-notifications';
import { ConfigProvider } from '../hooks/useConfig';
import { I18nProvider } from '../hooks/useI18n';
import { HelmetProvider } from 'react-helmet-async';
import { UIConfigProvider } from '../hooks/useUIConfig';

const AllTheProviders = ({ children }) => {
  return (
    <SWRConfig value={SwrFetcherConfig}>
      <ToastProvider>
        <ConfigProvider>
          <I18nProvider>
            <HelmetProvider>
              <UIConfigProvider>{children}</UIConfigProvider>
            </HelmetProvider>
          </I18nProvider>
        </ConfigProvider>
      </ToastProvider>
    </SWRConfig>
  );
};

const customRender = (ui, options: any = {}) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
