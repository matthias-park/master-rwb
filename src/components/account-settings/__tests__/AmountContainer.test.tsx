import React from 'react';
import { render, cleanup, fireEvent, screen } from '../../../utils/testUtils';
import AmountContainer from '../AmountContainer';
import renderer from 'react-test-renderer';

beforeEach(cleanup);

test('displays title correctly', async () => {
  const content = 'test title';
  const { getByTestId } = render(<AmountContainer amount="" title={content} />);
  expect(getByTestId('amount-title')).toHaveTextContent(content);
});

test('displays amount correctly', async () => {
  const content = '3,043.74 €';
  const { getByTestId } = render(<AmountContainer amount={content} title="" />);
  expect(getByTestId('amount-amount')).toHaveTextContent(content);
});

test('tooltip hidden', async () => {
  const content = 'tooltip';
  render(<AmountContainer amount="" title="" tooltip={content} />);
  expect(screen.queryByText(content)).not.toBeInTheDocument();
});

test('tooltip visible', async () => {
  const content = 'tooltip';
  const { getByTestId, findByText } = render(
    <AmountContainer amount="" title="" tooltip={content} />,
  );
  fireEvent.mouseOver(getByTestId('amount-tooltip'));
  expect(await findByText(content)).toBeInTheDocument();
});

test('matches snapshot', () => {
  const titleContent = 'test title';
  const amountContent = '3,043.74 €';
  const tooltipContent = 'tooltip';
  const dom = renderer.create(
    <AmountContainer
      title={titleContent}
      amount={amountContent}
      tooltip={tooltipContent}
    />,
  );
  expect(dom).toMatchSnapshot();
});
