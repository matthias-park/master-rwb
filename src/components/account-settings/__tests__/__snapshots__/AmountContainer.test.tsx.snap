// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="amount-container mb-4"
>
  <small
    className="amount-container__text"
    data-testid="amount-title"
  >
    test title
  </small>
  <h2
    className="amount-container__amount"
    data-testid="amount-amount"
  >
    3,043.74 â‚¬
  </h2>
  <i
    className="amount-container__icon icon-tooltip"
    data-testid="amount-tooltip"
    onBlur={[Function]}
    onFocus={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  />
</div>
`;
